packages/airnode-protocol/contracts/rrp/AirnodeRrp.sol

```diff
@@ -58,6 +58,8 @@ contract AirnodeRrp is
 
     /// @notice Called by the requester to make a request that refers to a
     /// template for the Airnode address, endpoint ID and parameters
+    /// @dev Validation of `sponsor` and `sponsorWallet` are left to the node.
+    /// `fulfillAddress` does not have to belong to a contract.
```

Comment:
I suspect it isn't possible to do it on-chain with the current `calldata` as it lacks the Airnode's xpub, but besides that it would also add additional computational complexity on-chain without any benefit (if the request is invalid, the node won't respond and the request will drift out of the logs retrieval window).
I personally don't think it's necessary to detail this in the comment/inline docs as if you're reading at this level you should be able to figure it out (if it was a question in the first place).

Reply:
I extended the docs (https://github.com/api3dao/airnode/pull/748/commits/085c68aa0f3627bb1d453ef1334646734632dc22) but the exact why is a subjective choice (as in you may have as easily require `fulfillAddress` to belong to a contract at request or fulfillment time)